### ISSUE TITLE
kfctl.sh delete all failed on OSX

### ISSUE THREAD

**Issue by IronPan** [kubeflow/kubeflow#2411]
Error:
```
...

++ expr match 'name: default
kubernetesversion: v1.11.6
path: default
destination:
  server: https://35.238.191.252
  namespace: kubeflow
targets: []
libraries: {}' '.*server[^\.0-9]*\([\.0-9]\+\)'
expr: syntax error
```

Line of code causing the issue 
https://github.com/kubeflow/kubeflow/pull/2305/files#diff-8a6f2d237f052b07ca594c20830ed9d2R474
This only happens on macbook which has different expr syntax 
ref https://stackoverflow.com/questions/25587446/syntax-error-expr

we should use some parser that are more standardized.

**Comment by jlewi** [comment]
We're moving to a go binary which should so solve this issue.

**Comment by jlewi** [comment]
Reopening this as it might be good to have a quick fix for MacOSX.

Is it possible to detect whether we are running on OSX and just skip the check?

**Comment by svalleru** [comment]
eh, regexes can get messy
@jlewi @IronPan here's the quick fix - #2453. please review.

**Comment by svalleru** [comment]
@jlewi @IronPan can you guys close this issue?
